pkg/tests/schema_pulumi_test.go

```diff
@@ -1508,3 +1509,263 @@ runtime: yaml
 		})
 	}
 }
+
+func TestConfigureGetRaw(t *testing.T) {
+	getOkExists := func(d *schema.ResourceData, key string) (interface{}, bool) {
+		v := d.GetRawConfig().GetAttr(key)
+		if v.IsNull() {
+			return nil, false
+		}
+		return d.Get(key), true
+	}
+	resMap := map[string]*schema.Resource{
+		"prov_test": {
+			Schema: map[string]*schema.Schema{
+				"test": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+		},
+	}
+
+	runConfigureTest := func(t *testing.T, configPresent bool) {
+		tfp := &schema.Provider{
+			ResourcesMap: resMap,
+			Schema: map[string]*schema.Schema{
+				"config": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+			ConfigureContextFunc: func(ctx context.Context, rd *schema.ResourceData) (interface{}, diag.Diagnostics) {
+				_, ok := getOkExists(rd, "config")
+				require.Equal(t, configPresent, ok, "Unexpected config value")
+				return nil, nil
+			},
+		}
+		bridgedProvider := pulcheck.BridgedProvider(t, "prov", tfp)
+		configVal := "val"
+		if !configPresent {
+			configVal = "null"
+		}
+		program := fmt.Sprintf(`
+name: test
+runtime: yaml
+resources:
+  prov:
+    type: pulumi:providers:prov
+	defaultProvider: true
+	properties:
+	  config: %s
+  mainRes:
+    type: prov:index:Test
+	properties:
+	  test: "hello"
+outputs:
+  testOut: ${mainRes.test}
+`, configVal)
+		pt := pulcheck.PulCheck(t, bridgedProvider, program)
+		res := pt.Up()
+		require.Equal(t, "hello", res.Outputs["testOut"].Value)
+	}
+
+	t.Run("config exists", func(t *testing.T) {
+		runConfigureTest(t, true)
+	})
+
+	t.Run("config does not exist", func(t *testing.T) {
+		runConfigureTest(t, false)
+	})
+}
+
+// TODO[pulumi/pulumi-terraform-bridge#2274]: Move to actual cross-test suite once the plumbing is done
+func TestConfigureCrossTest(t *testing.T) {
+	resMap := map[string]*schema.Resource{
+		"prov_test": {
+			Schema: map[string]*schema.Schema{
+				"test": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+		},
+	}
+
+	runTest := func(t *testing.T, sch map[string]*schema.Schema, pulumiProgram, tfProgram string) {
+		var tfRd *schema.ResourceData
+		var puRd *schema.ResourceData
+		_ = puRd // ignore unused warning
+		tfp := &schema.Provider{
+			ResourcesMap: resMap,
+			Schema:       sch,
+			ConfigureContextFunc: func(ctx context.Context, rd *schema.ResourceData) (interface{}, diag.Diagnostics) {
+				if tfRd == nil {
+					tfRd = rd
+				} else {
+					puRd = rd
+				}
+
+				return nil, nil
+			},
+		}
+
+		tfdriver := tfcheck.NewTfDriver(t, t.TempDir(), "prov", tfp)
+		tfdriver.Write(t, tfProgram)
+		tfdriver.Plan(t)
+		require.NotNil(t, tfRd)
+		require.Nil(t, puRd)
+
+		bridgedProvider := pulcheck.BridgedProvider(t, "prov", tfp)
+
+		pt := pulcheck.PulCheck(t, bridgedProvider, pulumiProgram)
+		pt.Preview()
+		require.NotNil(t, puRd)
+		require.Equal(t, tfRd.GetRawConfig(), puRd.GetRawConfig())
```

Comment:
This is very nice. This gives the coverage I was looking for - do we still need  TestConfigureGetRaw? Or that one just checks something new compared to this test matrix here?

Reply:
I added it because it uses the code used in the provider as a regression test. It's not strictly necessary but given that we'll move this code out of here at some point there's probably some value in having an explicit regression test for the panic.

LMK if you disagree, I don't feel strongly either way.

Reply:
Just a comment could help. LGTM.